cache-filter/src/filter/http.rs

```diff
@@ -241,17 +249,37 @@ impl CacheFilter {
             }
         }
 
-        let app = Application {
-            app_id: self.cache_key.app_id().clone(),
-            service_id: self.cache_key.service_id().clone(),
-            local_state: state,
-            metric_hierarchy: hierarchy,
-        };
+        let app;
+        if let Some(app_keys) = response.app_keys() {
+            let keys = app_keys
+                .keys()
+                .iter()
+                .map(|app_key| AppKey::from(app_key.as_ref()))
+                .collect::<Vec<_>>();
+            app = Application {
+                app_id,
+                service_id,
+                local_state: state,
+                metric_hierarchy: hierarchy,
+                app_keys: Some(keys),
+            };
+        } else {
+            app = Application {
+                app_id,
+                service_id,
+                local_state: state,
+                metric_hierarchy: hierarchy,
+                app_keys: None,
+            };
+        }
 
         set_application_to_cache(&self.cache_key.as_string(), &app, true, None);
 
         match self.handle_cache_hit(&RefCell::new(app)) {
-            Ok(()) => Ok(()),
+            Ok(()) => {
+                self.resume_http_request();
```

Comment:
Another great find! btw, this should be a part of `handle_cache_hit` flow. Can you please shift this to after line: 154 i.e if MQ doesn't fail.

Reply:
yeah i was actually going to put that there but in the case of a cache hit , it returns `Action::Continue` from the outer function. So that's why I handled it from outside (not to use `self.resume` and `Action::Continue` at the same time). But anyway there's no issue with using both `resume` and `continue` , it will take whoever comes first. For now I moved it into `handle_cache_hit` and maybe we can refactor things later.